src/components/UserProfile/UserProfile.jsx

```diff
@@ -15,367 +15,367 @@ import {
+  const [summarySelected, setSummarySelected] = useState(null)
+  const [summaryName, setSummaryName] = useState('')
+  const [showSummary, setShowSummary] = useState(false)
+  const [saved, setSaved] = useState(false)
+
+  const userProfileRef = useRef()
+
+  const isTasksEqual = JSON.stringify(originalTasks) === JSON.stringify(tasks)
+  const isProfileEqual = JSON.stringify(userProfile) === JSON.stringify(originalUserProfile)
+  const [codeValid, setCodeValid] = useState(true)
+
+  const [userStartDate, setUserStartDate] = useState('')
+  const [userEndDate, setUserEndDate] = useState('')
 
   /* useEffect functions */
   useEffect(() => {
-    loadUserProfile();
-    dispatch(fetchAllProjects());
-    dispatch(getAllUserTeams());
-  }, []);
+    loadUserProfile()
+    dispatch(fetchAllProjects())
+    dispatch(getAllUserTeams())
+  }, [])
 
   useEffect(() => {
-    userProfileRef.current = userProfile;
-  });
+    userProfileRef.current = userProfile
+  })
 
   useEffect(() => {
-    checkIsTeamsEqual();
-    checkIsProjectsEqual();
-    setUserProfile({ ...userProfile, teams, projects });
-    setOriginalUserProfile({ ...originalUserProfile, teams, projects });
-  }, [teams, projects]);
+    checkIsTeamsEqual()
+    checkIsProjectsEqual()
+    setUserProfile({ ...userProfile, teams, projects })
+    setOriginalUserProfile({ ...originalUserProfile, teams, projects })
+  }, [teams, projects])
 
   useEffect(() => {
-    setShowLoading(true);
-    loadUserProfile();
-    loadUserTasks();
-  }, [props?.match?.params?.userId]);
+    setShowLoading(true)
+    loadUserProfile()
+    loadUserTasks()
+  }, [props?.match?.params?.userId])
 
   useEffect(() => {
-    if (!blueSquareChanged) return;
-    setBlueSquareChanged(false);
-    handleSubmit();
-  }, [blueSquareChanged]);
+    if (!blueSquareChanged) return
+    setBlueSquareChanged(false)
+    handleSubmit()
+  }, [blueSquareChanged])
 
   const checkIsTeamsEqual = () => {
-    setOriginalTeams(teams);
-    const originalTeamProperties = [];
+    setOriginalTeams(teams)
+    const originalTeamProperties = []
     originalTeams?.forEach(team => {
       for (const [key, value] of Object.entries(team)) {
         if (key == 'teamName') {
-          originalTeamProperties.push({ [key]: value });
+          originalTeamProperties.push({ [key]: value })
         }
       }
-    });
+    })
 
-    const teamsProperties = [];
+    const teamsProperties = []
     teams?.forEach(team => {
       for (const [key, value] of Object.entries(team)) {
         if (key == 'teamName') {
-          teamsProperties.push({ [key]: value });
+          teamsProperties.push({ [key]: value })
         }
       }
-    });
+    })
 
     const originalTeamsBeingDisplayed = teamsProperties.filter(
       item =>
         JSON.stringify(item) ===
         JSON.stringify(originalTeamProperties.filter(elem => elem.teamName === item.teamName)[0]),
-    );
+    )
 
     const compare =
```

Comment:
There is no need to have repeated work; you can refactor into a function or variable.